### PR TITLE
feature(installer): update installer version and improve desktop version launcher

### PR DESCRIPTION
**Pull request by sylvlecl**
- updates installer submodule to benefit from last fixes and v2.18 handling
 - make application exit when the server process ends, instead of hanging
 - detect when server is already running
 - separate one popup for "starting" and another for "started"
 - fix alembic handling in pyinstaller:
   alembic must not be defined as a script, otherwise it's executed when the application
   quits. Instead, it must just be analyzed separately to get its dependencies

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - "master"
       - "hotfix/**"
+      - "feature/update-installer"
```
Comment (MartinBelthle) [inline .github/workflows/deploy.yml]: Should be removed or we could add more generic branch name such as `installer/**` or `deploy/**`

File: AntaresWebLinux.spec [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
-from PyInstaller.utils.hooks import collect_dynamic_libs
 
 block_cipher = None
 
-# We need to analyze all alembic files to be sure the migration phase works fine
-migrations_dir = Path('alembic/versions')
-migration_files = [str(f) for f in migrations_dir.iterdir() if f.is_file() and f.suffix == '.py']
+# We need to analyze all alembic files to be sure the migration phase works fine:
+# alembic loads version files by their path, so we need to add them as "data" to the package,
+# but all the dependencies they use need to be included also, wo we need to perform a
```
Comment (MartinBelthle) [inline AntaresWebLinux.spec:8]: typo: so we need

File: AntaresWebLinux.spec [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
 
-binaries = [('./alembic.ini', './alembic.ini')]  + collect_dynamic_libs('tables')
-
-antares_web_server_a = Analysis(['antarest/gui.py', 'alembic/env.py'] + migration_files,
+antares_web_server_a = Analysis(['antarest/gui.py'],
              pathex=[],
-             binaries=binaries,
-             datas=[('./resources', './resources'), ('./alembic', './alembic')],
+             binaries=[],
+             datas=[('./resources', './resources'), ('./alembic', './alembic'), ('./alembic.ini', './')],
```
Comment (MartinBelthle) [inline AntaresWebLinux.spec:17]: Why did you add `./` ?

File: AntaresWebWin.spec [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
-from PyInstaller.utils.hooks import collect_dynamic_libs
 
 block_cipher = None
 
-# We need to analyze all alembic files to be sure the migration phase works fine
-migrations_dir = Path('alembic/versions')
-migration_files = [str(f) for f in migrations_dir.iterdir() if f.is_file() and f.suffix == '.py']
+# We need to analyze all alembic files to be sure the migration phase works fine:
+# alembic loads version files by their path, so we need to add them as "data" to the package,
+# but all the dependencies they use need to be included also, wo we need to perform a
```
Comment (MartinBelthle) [inline AntaresWebWin.spec:8]: same typo

File: antarest/gui.py [hunk 5]
```diff
@@ -39,67 +41,139 @@ def run_server(config_file: Path) -> None:
...
+    open_app_action = menu.addAction("Open application")
+    assert open_app_action is not None
+    open_app_action.triggered.connect(open_app)
+    # To quit the app
+    quit_action = menu.addAction("Quit")
+    assert quit_action is not None
+    quit_action.triggered.connect(app.quit)
+
+    # Adding options to the System Tray
+    def handle_action(reason: int) -> None:
```
Comment (MartinBelthle) [inline antarest/gui.py]: This function/design is really weird but I'm far from understanding this file

**Review by MartinBelthle — DISMISSED** [review]
The `gui.py` file is really hard to understand IMO.
I wouldn't mind more doc cause I'm not familiar with this code :smiley:

File: AntaresWebLinux.spec (hunk 3 above)
Comment (sylvlecl) [reply AntaresWebLinux.spec:17]: Before we had this: `binaries = [('./alembic.ini', './alembic.ini')]` which created a folder called "alembic.ini" in which the file was placed.

Now the file is correctly placed at the root
